### PR TITLE
Leader App

### PR DESCRIPTION
**Pull request by gabryelreyes**
- Added platoon-specific configurations
- Added SerialMuxChannels for RemoteControl and data transfers, as well as their configuration in the App
- Separated module initialization (SMP, MQTT) into separate methods for readability and maintainability.
- Removed unnecessary components (Controllers, Safety policies, PlatoonController) from leader, as these only work on followers.
- Updated Waypoint class with constructors and utilities
- Added a LongitudinalController to calculate the top driving speed of the RU

### CODE REVIEW

File: lib/ConvoyLeader/src/App.cpp [hunk 1]
```diff
@@ -326,6 +319,81 @@ void App::fatalErrorHandler()
...
+{
+    /* Setup MQTT Server, Birth and Will messages. */
+    bool                                           isSuccessful = false;
+    SettingsHandler&                               settings     = SettingsHandler::getInstance();
+    StaticJsonDocument<JSON_BIRTHMESSAGE_MAX_SIZE> birthDoc;
+    char                                           birthMsgArray[JSON_BIRTHMESSAGE_MAX_SIZE];
+    String                                         birthMessage;
+
+    birthDoc["name"] = settings.getRobotName();
+    (void)serializeJson(birthDoc, birthMsgArray);
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.cpp:332]: Might go wrong.

File: lib/ConvoyLeader/src/App.cpp [hunk 2]
```diff
@@ -326,6 +319,81 @@ void App::fatalErrorHandler()
...
+    birthMessage = birthMsgArray;
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.cpp:333]: Hmm ... having already the birth message in an array and allocating it additional on the heap ... could we avoid it by changing the MqttSettings structure?

File: lib/ConvoyLeader/src/App.cpp [hunk 3]
```diff
@@ -326,6 +319,81 @@ void App::fatalErrorHandler()
...
+    else if (false == m_mqttClient.setConfig(mqttSettings))
+    {
+        LOG_FATAL("MQTT configuration could not be set.");
+    }
+    else
+    {
+        isSuccessful =
+            m_mqttClient.subscribe(TOPIC_NAME_RELEASE, true, [this](const String& payload) { this->release(); });
+
+        isSuccessful &= m_mqttClient.subscribe("platoons/0/vehicles/0/feedback", false,
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.cpp:357]: "&=" please do it explicit with if-else clause.

File: lib/ConvoyLeader/src/App.h [hunk 4]
```diff
@@ -100,48 +105,58 @@ class App
...
+     * @param[in] feedback Feedback from last follower.
      */
-    bool motorSetpointCallback(const int16_t left, const int16_t right);
+    void setLastFollowerFeedback(const Waypoint& feedback);
 
 private:
     /** Minimum battery level in percent. */
     static const uint8_t MIN_BATTERY_LEVEL = 10U;
 
+    /** Send waypoint timer interval. */
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.h:140]: Unit?

File: lib/ConvoyLeader/src/App.h [hunk 5]
```diff
@@ -161,9 +176,24 @@ class App
...
+
+    /**
+     * Longitudinal controller.
+     */
+    LongitudinalController m_longitudinalController;
+
+    /**
+     * Latest vehicle data from RU.
+     */
+    Waypoint latestVehicleData;
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.h:191]: "m_" is missing.

File: lib/ConvoyLeader/src/LongitudinalController.h [hunk 6]
```diff
@@ -53,60 +55,127 @@
...
+     * LongitudinalController constructor.
      */
     LongitudinalController();
 
     /**
-     * Longitudinal Controller destructor.
+     * Default destructor.
      */
-    virtual ~LongitudinalController();
+    ~LongitudinalController();
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/LongitudinalController.h:85]: Isn't it still the destructor?

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/ConvoyLeader/src/App.cpp [hunk 7]
```diff
@@ -215,33 +216,44 @@ void App::loop()
     /* Process Longitudinal Controller */
     m_longitudinalController.process();
 
-    if (false == m_initialDataSent)
+    if (true == m_initialCommandTimer.isTimeout())
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/App.cpp:219]: The whole block will later be part of an initial state, right?

File: lib/ConvoyLeader/src/RemoteControl.h [hunk 8]
```diff
@@ -65,6 +65,7 @@ namespace RemoteControl
         CMD_ID_REINIT_BOARD,            /**< Re-initialize the board. Required for webots simulation. */
         CMD_ID_GET_MAX_SPEED,           /**< Get maximum speed. */
         CMD_ID_START_DRIVING,           /**< Start driving. */
+        CMD_ID_SET_INIT_POS,            /**< Set initial position. */
```
Comment (BlueAndi) [inline lib/ConvoyLeader/src/RemoteControl.h:68]: Remove last ",".

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: lib/ConvoyLeader/src/App.cpp (hunk 7 above)
Comment (gabryelreyes) [reply lib/ConvoyLeader/src/App.cpp:219]: Yes!

Comment (BlueAndi) [reply lib/ConvoyLeader/src/App.cpp:219]: Ok.

**Pull request closed without merging**